test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,34 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_info_with_bad_args) {
+  rmw_ret_t ret;
```

Comment:
Suggestion: move to line 388 where ret is used for the first time

Reply:
But all 5 variables get used on the same line (388), moving one closer would get the other further

Reply:
Why is that a problem?

Reply:
The problem is that @Blast545 pointed out 2 variables, the 2 further, to move them closer. Whenever I move those 2 closer, the other 2 are going to get moved away from the first place they are used, and the problem will just loop around, as there are 5 variables that get used for the first time in the same place. My question would be, are some variables more important to be closer than the others?

Reply:
The first variable, `rmw_ret_t ret` can be initialized in the same place where it is used, by moving it there you are saving a declaration.

As for the second one, `bool taken` yeah, you are right and there is no need to move it. But add an initialization value to its declaration

Reply:
Updated